### PR TITLE
Feature internal channels

### PR DESCRIPTION
**Pull request by VivaanKhatri**
## Proposed Changes
This implementation allows the connection of an inlet surface with an outlet surface (e.g. to simulate internal channels inside a model). In order to do so, it has been necessary to create an option OUTLET_TYPE that will be used in the future (e.g. to add a mass flow driver for the outlet - currently only static pressure is implemented - ).

To sum up, to connect an inlet with an outlet you should type the following:

% Inlet boundary type (TOTAL_CONDITIONS, MASS_FLOW, SURFACE_OUTLET)
 INLET_TYPE= SURFACE_OUTLET
% Outlet boundary type (PRESSURE_OUTLET, SURFACE_INLET)
OUTLET_TYPE= SURFACE_INLET

In the future, we should be able to specify pressure losses inside the "pipe" that connect the surfaces.
 
## Related Work
This is a new implementation that solves passive flow control problems where we are not interested in creating a complicated mesh inside the main volume to define the internal channels/pipes.

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [X] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.

### CODE REVIEW

File: SU2_CFD/src/output_structure.cpp [hunk 1]
```diff
@@ -4579,13 +4579,6 @@ void COutput::SetConvHistory_Body(ofstream *ConvHist_file,
 
           /*--- For specific applications, evaluate and plot the surface. ---*/
 
-          if (config[val_iZone]->GetnMarker_Analyze() != 0) {
```
Comment (rsanfer) [inline SU2_CFD/src/output_structure.cpp]: Why is this removed? Is SpecialOutput_Distortion not working for incompressible? (That's the only difference with the if statement below).

File: SU2_CFD/src/solver_direct_mean.cpp [hunk 2]
```diff
@@ -11218,17 +11218,30 @@ void CEulerSolver::BC_Inlet(CGeometry *geometry, CSolver **solver_container,
 
         /*--- Total properties have been specified at the inlet. ---*/
 
-        case TOTAL_CONDITIONS:
-
+        case TOTAL_CONDITIONS: case SURFACE_OUTLET:
+          
           /*--- Retrieve the specified total conditions for this inlet. ---*/
-
+
```
Comment (rsanfer) [inline SU2_CFD/src/solver_direct_mean.cpp]: Same as above, please remove extra whitespace.

File: SU2_CFD/src/solver_direct_mean.cpp [hunk 3]
```diff
@@ -11218,17 +11218,30 @@ void CEulerSolver::BC_Inlet(CGeometry *geometry, CSolver **solver_container,
...
           else P_Total  = config->GetInlet_Ptotal(Marker_Tag);
           T_Total  = config->GetInlet_Ttotal(Marker_Tag);
           Flow_Dir = config->GetInlet_FlowDir(Marker_Tag);
+          
+          if ((Kind_Inlet == SURFACE_OUTLET) && (config->GetExtIter() != 0)) {
+            
+            /*--- Retrieve the specified total conditions for this inlet. ---*/
+            
+            if (gravity) P_Total = config->GetSurface_TotalPressure(0) - geometry->node[iPoint]->GetCoord(nDim-1)*STANDARD_GRAVITY;
+            else P_Total  = config->GetSurface_TotalPressure(0);
```
Comment (rsanfer) [inline SU2_CFD/src/solver_direct_mean.cpp]: Why is the Marker_Tag in the calls 

- config->GetSurface_TotalPressure(0)
- config->GetSurface_TotalTemperature(0)

forced to be 0?

**Comment by economon** [comment]
@VivaanKhatri : any news or updates to this PR? It has been sitting for some time.. looks to be a useful contribution and shares some options/keywords with #621 (hopefully not too conflicting).

**Comment by rsanfer** [comment]
This has been inactive for a while and there are some conflicts with develop, so I'll close it for now. However, it is a very interesting feature, so I would be happy to see this resubmitted soon with some additional fixes.
